### ISSUE TITLE
Have water shadows and caustics render through transparent blocks

### ISSUE THREAD

**Issue by marlpur** [X0nk/Bliss-Shader#320]
I've noticed that unlike some other shader packs, Bliss does not render water shadows and caustics if the water is behind a transparent block like glass, instead light is just rendered as if it were passing only through the glass.

Would it be possible to add this behaviour to the shader as I love the caustics in this pack and want to be able to create underwater rooms with glass walls.

**Comment by marlpur** [comment]
Okay I've done a bit more investigation of this behaviour and it seems that:

Water shadows and caustics are not cast through translucent blocks when the player is not underwater.
Water shadows and caustics are only cast through translucent blocks if they player is underwater or looking through water.
Water shadows and caustics are cast on blocks directly below the water source rather than being angled with the sun/moon.

I'm not an expert on shaders but it seems some other packs handle this by having the water cast a shadow and having the caustics be part of the shadow map, could this be implemented somehow in Bliss?

I tried editing the code in composite1.fsh but could not get the caustics to be visible when not in water or looking though it.

**Comment by X0nk** [comment]
yeah, the caustics are not advanced or anything. i just check if the current pixel is water/if youre under water, and if it is, do that effect on those pixels.

Doing what you want is a bit more complicated, and i wanted to try and add it sometimes. it just looks too cool in like seus renewed/ptgi for example
